identity/src/commonMain/kotlin/com/android/identity/nfc/NfcIsoTag.kt

```diff
@@ -0,0 +1,218 @@
+                payload = ByteString(),
+                le = length
+            )
+        )
+        if (response.status != Nfc.RESPONSE_STATUS_SUCCESS) {
+            throw IllegalStateException("Error READ BINARY, status ${response.statusHexString}")
+        }
+        return response.payload.toByteArray()
+    }
+
+    /**
+     * Updates binary data according to ISO 7816-4 clause 11.3.5.
+     *
+     * @param offset offset of where to update data.
+     * @param data the data to write, cannot be larger than 255 bytes.
+     */
+    suspend fun updateBinary(offset: Int, data: ByteArray) {
+        require(data.size > 0) { "Data to write must be non-empty" }
+        require(data.size < 256) { "Data cannot be larger than 255 bytes" }
+
+        //
+        val response = transceive(
+            CommandApdu(
+                cla = 0,
+                ins = 0xd6,
+                p1 = offset / 0x100,
+                p2 = offset.and(0xff),
+                payload = ByteString(data),
+                le = 0
+            )
+        )
+        if (response.status != Nfc.RESPONSE_STATUS_SUCCESS) {
+            throw IllegalStateException("Error UPDATE BINARY, status ${response.statusHexString}")
+        }
+    }
+
+    /**
+     * Reads NDEF data according to NFC Forum Type 4 Tag section 7.5.4.
+     *
+     * @param wtInt Minimum waiting time as per NFC Forum Tag NDEF Exchange Protocol section 4.1.6.
+     * @param nWait Maximum number of waiting time extensions as per NFC Forum Tag NDEF Exchange Protocol section 4.1.7.
+     * @return the message that was read.
+     */
+    suspend fun ndefReadMessage(
+        wtInt: Int = 0,
+        nWait: Int = 0
+    ): NdefMessage {
+        var nWait_ = nWait
+        var replyLen: Int
+        do {
+            replyLen = decodeShort(readBinary(0x0000, 2))
+            if (replyLen > 0) {
+                break
+            }
+
+            // As per [TNEP] 4.1.7 if the tag sends an empty NDEF message it means that
+            // it's requesting extra time... honor this if we can.
+            if (nWait_ > 0) {
+                val tWait = Duration.fromWtInt(wtInt)
+                delay(tWait)
+                nWait_--
+            } else {
+                throw IllegalStateException("NDEF message with length 0 but no time extensions left")
+            }
+        } while (true)
+        return NdefMessage.fromEncoded(readBinary(2, replyLen))
+    }
+
+    /**
+     * Exchanges NDEF messages according to NFC Forum Tag NDEF Exchange Protocol section 5.
+     *
+     * @param ndefMessage the message to write.
+     * @param wtInt Minimum waiting time as per NFC Forum Tag NDEF Exchange Protocol section 4.1.6.
+     * @param nWait Maximum number of waiting time extensions as per NFC Forum Tag NDEF Exchange Protocol section 4.1.7.
+     * @return the message which was read.
+     */
+    suspend fun ndefTransact(
+        ndefMessage: NdefMessage,
+        wtInt: Int,
```

Comment:
Same question as ndefReadMessage, regarding parameter names and types.

Reply:
yeah, same answer, I think I prefer to keep it this way.